examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -3,77 +3,268 @@
+# noise is skewed towards high values (large outliers). Also note that the
+# median estimate is smoother because of its natural robustness to outliers.
+#
+# Also observe that the inductive bias of gradient boosting trees is
+# unfortunately preventing our 0.05 quantile to fully capture the sinoisoidal
+# shape of the signal, in particular around x=8. Tuning hyper-parameters can
+# reduce this effect as shown in the last part of this notebook.
+#
+# Analysis of the error metrics
+# -----------------------------
+#
+# Measure the models with :func:`mean_square_error` and :func:`pinball_loss`
+# metrics on the training dataset.
+from pandas import DataFrame
 
-# Make the prediction on the meshed x-axis
-y_lower = clf.predict(xx)
+results = []
+for name, gbr in sorted(all_models.items()):
+    metrics = {'model': name}
+    y_pred = gbr.predict(X_train)
+    for alpha in [0.05, 0.5, 0.95]:
+        metrics["pbl=%1.2f" % alpha] = pinball_loss(
+            y_train, y_pred, alpha=alpha)
+    metrics['MSE'] = mean_squared_error(y_train, y_pred)
+    results.append(metrics)
+DataFrame(results).set_index('model')
 
-clf.set_params(loss='ls')
-clf.fit(X, y)
+# %%
+# One column shows all models evaluated by the same metric. The minimum number
+# on a column should be obtained when the model is trained and measured with
+# the same metric. This should be always the case on the training set if the
+# training converged.
+#
+# Note that because the target noise is skewed by the presence of large
+# outliers, the expected conditional mean and conditional median are
+# signficiantly different and therefore one could not use the least squares
+# model get a good estimation of the conditional median nor the converse.
+#
+# If the target distribution were symmetric and had no outliers (e.g. with
+# a Gaussian noise), then median estimator and the least squares estimator
+# would have yielded similar predictions.
+#
+# We then do the same on the test set.
+results = []
+for name, gbr in sorted(all_models.items()):
+    metrics = {'model': name}
+    y_pred = gbr.predict(X_test)
+    for alpha in [0.05, 0.5, 0.95]:
+        metrics["pbl=%1.2f" % alpha] = pinball_loss(
+            y_test, y_pred, alpha=alpha)
+    metrics['MSE'] = mean_squared_error(y_test, y_pred)
+    results.append(metrics)
+DataFrame(results).set_index('model')
+
+# %%
+# Errors are higher meaning the models slightly overfitted the data. It still
+# shows the minimum of a metric is obtained when the model is trained by
+# minimizing this same metric.
+#
+# Note that the conditional median estimator is competitive with the least
+# squares estimator in terms of MSE on the test set: this can be explained by
+# the fact the least squares estimator is very sensitive to large outliers
+# which can cause significant overfitting. This can be seen on the right hand
+# side of the previous plot. The conditional median estimator is naturally
+# robust to outliers and overfits less.
```

Comment:
Competitive yes. But as already mentioned earlier, the median is a very biased estimate for the mean in this case, i.e. it underestimates (only exception arount x=2).

Reply:
I rephrased the sentence to mention the bias.